src/shared/components/community/community.tsx

```diff
@@ -647,6 +647,12 @@ export class Community extends Component<
     const blockCommunityRes = await HttpService.client.blockCommunity(form);
     if (blockCommunityRes.state == "success") {
       updateCommunityBlock(blockCommunityRes.data);
+      this.setState(s => {
+        if (s.communityRes.state == "success") {
+          s.communityRes.data.community_view.blocked =
+            blockCommunityRes.data.blocked;
+        }
+      });
     }
   }
```

Comment:
One thing I'd like to call out is that there is some duplication going on between here and [post.tsx](https://github.com/LemmyNet/lemmy-ui/pull/1316/files#diff-630788fba7307c5ca6e02c2ecd35085a0e801fe27e318bdfc0a5fd258b3e4e0aR730).

I'd like to pull this out in a separate PR and was wondering what the suggested way to do this is. I'm not a huge fan of large `util` libs since they can be pretty disorganized and tend to have pretty inconsistent interfaces. I have a separate branch where I pull this logic out into `UserService` that handles blocking communities and accepts an optional callback for updating component state.

```ts
  public static async followCommunity(
    form: FollowCommunity,
    updateCommunity?: (res: RequestState<CommunityResponse>) => void
  ): Promise<void> {
    /**
     * Sends a request to follow a community
     *
     * @param form - Form data for following a community
     * @callback updateCommunity - Optional callback for updating community state
     */
    const followCommunityRes = await HttpService.client.followCommunity(form);
    updateCommunity !== undefined && updateCommunity(followCommunityRes);

    // Update myuserInfo
    if (followCommunityRes.state == "success") {
      const communityId = followCommunityRes.data.community_view.community.id;
      const mui = UserService.Instance.myUserInfo;

      if (mui) {
        mui.follows = mui.follows.filter(i => i.community.id != communityId);
      }
    }
  }
```

Which would simplify these functions to something like

```ts
handleBlockCommunity(form) {
  UserService.blockCommunity(form, this.updateCommunity);
}
```

Reply:
Ah, that's great to hear. Inferno really hasn't aged well. Let me know if you'd like any help with the new UI. Is the idea to only migrate the frontend to leptos?